mask/eolearn/mask/cloud_mask.py

```diff
@@ -385,12 +384,14 @@ def _do_single_temporal_cloud_detection(self, bands):
     def _do_multi_temporal_cloud_detection(self, bands: np.ndarray, is_data: np.ndarray, sigma: float) -> np.ndarray:
         """Performs a cloud detection process on multiple scenes at once"""
         multi_proba = np.empty(np.prod(bands.shape[:-1]))
-        img_size = np.prod(bands.shape[1:-1])
+        img_size = int(np.prod(bands.shape[1:-1]))
 
         n_times = bands.shape[0]
 
-        local_avg, local_var = None, None
-        prev_left, prev_right = None, None
+        local_avg: Optional[np.ndarray] = None
```

Comment:
What are the pros/cons of changing the type to include the Optional nature here, instead of in the function definition?

Reply:
could you elaborate?

Reply:
nevermind, I'm blind.